web-console/src/utils/druid-query.tsx

```diff
@@ -43,3 +43,61 @@ export async function queryDruidSql(sqlQuery: Record<string, any>): Promise<any[
   }
   return sqlResultResp.data;
 }
+
+function parseQueryRel(queryRel: string) {
+  if (!queryRel) {
+    return {
+      query: null,
+      signature: null
+    };
+  }
+
+  const queryAndSignature = queryRel.split(', signature=');
+  const queryValue = new RegExp(/query=(.+)/).exec(queryAndSignature[0]);
+  const signatureValue = queryAndSignature[1];
+
+  let parsedQuery: any;
+
+  if (queryValue && queryValue[1]) {
+    try {
+      parsedQuery = JSON.parse(queryValue[1]);
+    } catch (e) {}
+  }
+
+  return {
+    query: parsedQuery || queryRel,
+    signature: signatureValue || null
+  };
+}
+
+export function parseQueryPlan(raw: string): any {
+  let plan: string = raw;
+
+  plan = plan.replace('\n', '');
```

Comment:
Are you sure this is correct? This will only replace the first occurrence. Try it with a couple of string in the browser.